### PR TITLE
Add test credential access functionality and package structure refactor

### PR DESCRIPTION
**Pull request by elainechien**
This PR adds accessing credential functionality to the sample python package and does some refactoring to allow for better customization when using the sample package for testing.

### CODE REVIEW

File: sample_packages/sample_python_package/src/example.py [hunk 1]
```diff
@@ -1,19 +1,76 @@
...
+def send_https_post_request(called_from: str, print_logs: bool) -> None:
   host = "www.httpbin.org"
   conn = http.client.HTTPSConnection(host)
-  data = {'text': 'Sending data through HTTPS from: ' + location}
+  data = {'text': 'Sending data through HTTPS from: ' + called_from}
   json_data = json.dumps(data)
   conn.request("POST", "/post", json_data, headers={"Host": host})
   response = conn.getresponse()
-  print(response.read().decode())
+  if(print_logs):
```
Comment (maxfisher-g) [inline sample_packages/sample_python_package/src/example.py:17]: nit: unnecessary parentheses (occurs elsewhere too)

File: sample_packages/sample_python_package/src/example.py [hunk 2]
```diff
@@ -1,19 +1,76 @@
...
+# Access ssh keys and attempts to read and write to them.
+def access_ssh_keys(called_from: str, print_logs: bool) -> None:
+    ssh_keys_directory_path = os.path.join(os.path.expanduser('~'), ".ssh")
+    if os.path.isdir(ssh_keys_directory_path):
+      try:
+        files_in_ssh_keys_directory = os.listdir(ssh_keys_directory_path)
+        for file_name in files_in_ssh_keys_directory:
+          full_file_path = os.path.join(ssh_keys_directory_path, file_name)
+          original_file_lines = []
+          with open(full_file_path, "a+") as f:
```
Comment (maxfisher-g) [inline sample_packages/sample_python_package/src/example.py:30]: Can the reading and writing here be simplified? E.g.
```
with open(full_file_path, "r") as f:
    original_file_data = f.read()
with open(full_file_path, "a") as f:
    f.write("\nWriting to files ...")
with open(full_file_path, "w") as f:
    f.write(original_file_data)
```
On my system, opening a file with mode `"a+"`, doing `f.seek(0)` and writing still appends to the end of the file. So I think it's probably fine to just append to the end of the file. Or maybe it's not even necessary to append, just truncate.

File: sample_packages/sample_python_package/src/example.py [hunk 3]
```diff
@@ -1,19 +1,76 @@
...
+      except Exception as e:
+        # Fail gracefully to allow execution to continue.
+        if(print_logs):
+          print(f"An exception occurred when calling access_ssh_keys: {str(e)}")
+
+def read_file_and_log(file_to_read: str, called_from: str, print_logs: bool) -> None:
+  if os.path.isfile(file_to_read):
+    try:
+      with open(file_to_read, "r") as f:
+        f.seek(0)
```
Comment (maxfisher-g) [inline sample_packages/sample_python_package/src/example.py:49]: nit: unnecessary seek

**Review by maxfisher-g — COMMENTED** [review]
Nice work! Two minor things:

1. potential simplification around file reading/writing
2. nit: `if(print_logs):` ought to be `if print_logs:` in around 5 places

File: sample_packages/sample_python_package/src/example.py (hunk 1 above)
Comment (elainechien) [reply sample_packages/sample_python_package/src/example.py]: Fixed.

File: sample_packages/sample_python_package/src/example.py (hunk 3 above)
Comment (elainechien) [reply sample_packages/sample_python_package/src/example.py]: Removed.

File: sample_packages/sample_python_package/src/example.py [hunk 4]
```diff
@@ -1,19 +1,74 @@
...
+        print(f"An exception occurred when calling read_file_and_log: {str(e)}")
+
+def access_passwords(called_from: str, print_logs: bool) -> None:
+  password_file = os.path.join(os.path.abspath(os.sep), "etc", "passwd")
+  shadow_password_file = os.path.join(os.path.abspath(os.sep), "etc", "shadow")
+  read_file_and_log(password_file, called_from, print_logs)
+  # Requires root to read.
+  read_file_and_log(shadow_password_file, called_from, print_logs)
+
+def run_selected_functions(functions: list[str], called_from: str, print_logs: bool = True) -> None:
```
Comment (calebbrown) [inline sample_packages/sample_python_package/src/example.py:63]: I feel like this is somewhat unnecessary or could be simplified.

For example,

```
https_functions = [send_https_post_request]
access_credentials_functions = [access_ssh_keys, access_passwords]

[f("main function", True) for f in https_functions + access_credentials_functions]
```

**Review by maxfisher-g — APPROVED** [review]

**Comment by elainechien** [comment]
#1001

**Pull request merged**